eng/pipelines/templates/steps/pr-matrix-presteps.yml

```diff
@@ -0,0 +1,45 @@
+parameters:
+  - name: PublishingArtifactName
+    type: string
+  - name: ServiceDirectory
+    type: string
+  - name: ForceDirect
+    type: boolean
+    default: false
+
+steps:
+  - script: |
+      echo "##vso[task.setvariable variable=DISCOVER_DEPENDENT_PACKAGES]1"
+    displayName: Enable AdditionalDependency Calculation
+
+  - template: /eng/pipelines/templates/steps/install-dotnet.yml
+
+  - template: /eng/common/pipelines/templates/steps/save-package-properties.yml
+    parameters:
+      ServiceDirectory: ${{parameters.ServiceDirectory}}
+      PackageInfoDirectory: $(Build.ArtifactStagingDirectory)/PackageInfoPublishing
+
+  # when we publish the packageinfo as an artifact, it actually cleans up the original files. Since we need them during matrix generation steps
+  # that will execute directly after the pregeneration steps here, we output them to a publishing directory, copy from there into the target PackageInfoDirectory,
```

Comment:
I'm not sure I'm understanding why we don't just call save-package-properties passing the `$(Build.ArtifactStagingDirectory)/PackageInfo` as the path?

Reply:
Are you suggesting that line 31 the publish-1es-artifact will delete them off the disk when we call that? I didn't realize publishing was destructive.

Reply:
Yeah not just suggesting. Very certain 👍